File: apps/JSApp/AppController.js
/* 
 * AppController.js
 * vienna
 * 
 * Created by Adam Beynon.
 * Copyright 2009 Adam Beynon.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

include('vienna/vienna');

var AppController = NSObject.extend({
     
    /**
        @type Integer
    */
    _testValue: 10,
   
    /**
        @outlet
        @type NSWindow
    */
    _window: IBOutlet(),
   
   /**
        @outlet
        @type NSArrayController
   */
   _arrayController: IBOutlet(),
   
    /**
        @outlet
        @type NSArray
    */
    _tableContent: IBOutlet(),
    
    _tempData: null,
    
    _tableSelections: null,
   
    init: function() {
        this._super();
       
        this._tempData = [
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" },
            { "name": "Adam", "age": 23, "band": "Led Zepplin" },
            { "name": "Benjamin", "age": 22, "band": "Rage Against the machine" },
            { "name": "Rebeccae", "age": 19, "band": "Lagy Gaga" }
        ];
       
       console.log('creating app controller');
       return this;
   },
   
   setTestValue: function(aValue) {
       this._testValue = aValue;
   }.property('testValue'),
   
   doSomething: function(sender) {
       
   },
   
   awakeFromNib: function(sender) {
       console.log("Awoken from nib");
   },
   
   applicationWillFinishLaunching: function() {
       console.log('Application will finish launching');
   },
   
   applicationDidFinishLaunching: function() {
       console.log("Application finished lauchiong");
   },

	/**
		Table view delegate
	*/
	numberOfRowsInTableView: function(tableView) {
		return this._tempData.length;
	},
	
	tableViewObjectValueForTableColumnRow: function(tableView, tableColumn, row) {
        // console.log(tableColumn);
	    return this._tempData[row][tableColumn.identifier()];
	}
});

File: apps/RubyWebApp/lib/controllers/app_controller.rb
# 
# app_controller.rb
# vienna
# 
# Created by Adam Beynon.
# Copyright 2009 Adam Beynon.
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
# 
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.
#

module RubyWebApp
  
  class AppController
    
    TABLE_VIEW_DATA = [
      { :name => 'Adam', :age => 23, :band => 'Led Zepplin' },
      { :name => 'Ben', :age => 20, :band => 'Pendulum' },
      { :name => 'Tom', :age => 30, :band => 'Tweenies' },
      { :name => 'Becky', :age => 12, :band => '50 pence' },
      { :name => 'Dad', :age => 24, :band => 'Take That' },
      { :name => 'Mum', :age => 25, :band => 'Rod Stewart' }
    ]
    
    def initialize
      # puts 'initialising app controller'
      @adam = 10
      @test_binding = false
    end
    
    def number_of_rows_in_table_view(table_view)
      TABLE_VIEW_DATA.length
    end
    
    def table_view(table_view, object_value_for_table_column:table_column, row:row)
      # TABLE_VIEW_DATA[row][table_column.identifier]
    end
    
    def adam?
      @adam
    end
    
    def test_binding
      @test_binding
    end
    
    def test_binding= aValue
      @test_binding = aValue
    end

    def will_finish_launching (notification)
      # puts 'Application will finish launching!'
    end

    def did_finish_launching (notification)
      # puts 'Application did finish launching!!'
      # JSON.get('http://search.twitter.com/trends.json?callback=vn_jsonp_callback_0', nil) do |json|
      #         puts json
      #         json[:trends].each do |trend|
      #           puts trend[:name]
      #         end
      #         # json is either an object with the result from the json feed, or, nil. nil is likely if an 
      #         # error occured
      #       end
    end
  end
end
